File: HashMap/1319-unique-number-of-occurrences/unique-number-of-occurrences.kt
class Solution {
    fun uniqueOccurrences(arr: IntArray): Boolean {
        val map = HashMap<Int,Int>()

        for(i in arr.indices){
            map[arr[i]] = map.getOrDefault(arr[i],0)+1
        }

        val list = mutableListOf<Int>()
        val set = HashSet<Int>()
        for(i in map.values){
            list.add(i)
            set.add(i)
        }

        if(list.count() == set.count()){
            return true
        }
        return false

    }
}

File: HashMap/1319-unique-number-of-occurrences/unique-number-of-occurrences.swift
class Solution {
    func uniqueOccurrences(_ arr: [Int]) -> Bool {
          var map = [Int: Int]()

        for num in arr {
            map[num, default: 0] += 1
        }

        var list = [Int]()
        var set = Set<Int>()
        for value in map.values {
            list.append(value)
            set.insert(value)
        }

        return list.count == set.count
    }
}